processor/src/advice/recorder.rs

```diff
@@ -0,0 +1,218 @@
+use super::{
+    AdviceInputs, AdviceProvider, AdviceSource, DataRecorder, ExecutionError, Felt, IntoBytes,
+    MerklePath, NodeIndex, RecordingMerkleStore, StarkField, Vec, Word,
+};
+
+// MEMORY ADVICE PROVIDER
+// ================================================================================================
+
+/// An in-memory `[AdviceProvider]` implementation to support program execution with support for
+/// data access recording. The recorder can be converted into a proof which can be used to provide
+/// the non-deterministic inputs for program execution.
+///
+/// Uses [DataRecorder] and [RecordingMerkleStore] as data backends.
+#[cfg(not(any(test, feature = "internals")))]
+#[derive(Debug, Clone, Default)]
+pub struct AdviceRecorder {
```

Comment:
I wonder if a better name for this would be `RecAdviceProvider`. This would be a bit more consistent with the naming of `MemAdviceProvider`.

Reply:
updated